cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1690,6 +1751,8 @@ func (r machineResource) provisioningEventForMachine(machineID string, e v1.Mach
 	} else if event.Event == metal.ProvisioningEventPhonedHome && len(ec.Events) > 0 && ec.Events[0].Event == metal.ProvisioningEventPhonedHome {
 		zapup.MustRootLogger().Sugar().Debugw("swallowing repeated phone home event", "id", ec.ID)
 		ec.Liveliness = metal.MachineLivelinessAlive
+	} else if event.Event == metal.ProvisioningEventReinstallAborted {
+		r.machineAbortReinstall(machineID)
```

Comment:
This is kind of a minor nit: I am not sure when "aborting the reinstall" happens, but I would let it call the metal-core directly on a dedicated endpoint and not indirectly via a machine event. This endpoint was built in a way that it does only deal with the event table. Now it's introducing more complex things.